zeebe/gateway-protocol/src/main/proto/rest-api.yaml

```diff
@@ -4017,7 +4017,7 @@ components:
     DocumentReference:
       type: object
       properties:
-        documentType:
+        camunda.document.type:
```

Comment:
> Why this naming format ?

We use this field to detect that a process variable contains a document and apply custom deserialization logic. The strange naming format is chosed deliberately to avoid accidental clashes with potential customer data - e.g. there is a high chance that any business payload contains something like `documentType`, so we shouldn't rely on a discriminator field with such a common name. Dots are meant to represent the "namespace". More details on the doc reference [here](https://docs.google.com/document/d/1fe4nYImFqCqGV1y0MXG4xe4a3U7Nb3cq_1n8OgJTSjk/edit#heading=h.ydnzsxjw8ev1).

> Why we need this enum ? Since is always set to camunda

There is a plan to support another document type `external` in the future - for documents that are not managed by Camunda but can be resolved/accessed by Camunda.